Mitigate duplicate active queries appearing in the Queries sidebar

MrDoomBringer opened the pull request:
This issue resolves `JIRA-NEBULA-743`

The Client Dev Tools would ometimes show duplicate active queries in the left side-panel. I believe this is because when getting queries it would manually read from the private `queryManager.queries` field, which explains the unexpected behavior we've been getting.

I've changed the logic to instead use the `queryManager.getObservableQueries()` function, which is supported and does exactly what we're trying to do here (i.e., get a list of active queries)
Before:
![image](https://user-images.githubusercontent.com/29008542/193101840-ff167d05-1ec7-4fa1-b5b5-60e9faae37cf.png)

After:
![image](https://user-images.githubusercontent.com/29008542/193101707-cc5a88a1-ebbd-4e33-bbe5-c9edd6d3bb2f.png)

mayakoneval commented:
@MrDoomBringer I am getting `<whatever>.getObservableQueries is not a function` as a type error when opening the local colors app that comes with the repo & trying to use apollo client. When console logging the `hook.ApolloClient` object, I see `queries` but no `getObservableQueries`

alex-wilmer reviewed (dismissed):
This is definitely an improvement - we need to investigate duplicate queries in the `queryManager` as well - but this should be okay to merge

mayakoneval commented:
@MrDoomBringer just adding the context here too: 

> we discovered that getObservableQueries only exists on some AC version between 3.3 & 3.6 and up. How do yall feel about serving the slightly bad duplicate query version to folks on the older versions of AC?
Emmanuel will add a check to see if the function exists & use one or the other depending. This should fix :team-michelle: 's demo (edited) 

also - can we update the `broadcastQueries` function to share logic here 🙏 would be great if we could put it in one place

mayakoneval commented:
@MrDoomBringer also curious if folks on your team have historical context for `getObservableQueries` - can we confirm the version it was added in?

MrDoomBringer commented:
> @MrDoomBringer also curious if folks on your team have historical context for `getObservableQueries` - can we confirm the version it was added in?

Yup! This was added in https://github.com/apollographql/apollo-client/pull/7813, `@apollo/client@3.4.0-rc.4`

mayakoneval commented:
@MrDoomBringer ping me when you need a re-review please!

MrDoomBringer commented:
@mayakoneval Should be good for a re-review. Updated broadcastQueries.js to use client.getObservableQueries() as well, and added compatibility checks for client versions <` 3.4.0`

MrDoomBringer commented:
>  I am unable to test the broadcastQueries code at all, I wonder if we should just not change that file righ tnow?

I was having trouble testing that specific code too, I agree that it should be fine to leave it for now. Thanks!
